surveys/2023-annual-survey/questions.md

```diff
@@ -0,0 +1,769 @@
+
+Type: select one (optional)
+
+1.76 (nightly), 1.75 (nightly), and then every version from 1.74 to 1.0 in descending order, and "a pre-1.0 version"
+
+> **justification**
+>
+> To get real data on how many people use older versions of the toolchain to
+> inform discussion on MSRVs.
+
+### Which editor or IDE setup do you use with Rust code on a regular basis?
+
+Type: select all that apply (optional)
+
+- VS Code
+- vi/vim/neovim
+- IntelliJ/CLion/other JetBrains IDE + Rust plugin
+- Rust Rover (dedicated IntelliJ Rust IDE)
+- Emacs
+- Sublime Text
+- Visual Studio
+- Xcode
+- Atom
+- Other (open response)
+
+> **justification**
+>
+> It is good to know which editor is the most preferred for Rust development. This
+> can change investment strategies for further IDE development.
+>
+> Note: previously this question included different 'drivers' of the Rust IDE
+> experience (e.g., racer, rls, rust-analyzer). Development has consolidated on
+> rust-analyzer, and so it's not necessary to find out which is being used.
+> If we are curious how far along adoption of rust-analyzer is, we can ask that
+> in a separate question, though this is likely easier to find out through download
+> numbers.
+
+## The Rust community
+
+### Roughly how often do you participate in the Rust community?
+
+Type: matrix (optional)
+
+Activities:
+
+- Produce informational content about Rust (e.g., blogged, live streamed, made a YouTube video, presented at a conference/meetup, etc.)
+- Consume informational content about Rust (e.g., blogs, live streams, YouTube videos, etc.)
+- Read official Rust communication channels (e.g., This Week in Rust, the official Rust blog, the Rust Twitter account, etc.)
+- Participate in conversations about Rust on social media or websites (Hacker News, r/rust, Twitter, LinkedIn, etc.)
+- Attend a Rust meetup or conference (virtual or in-person)
+- Write, comment on, contribute to discussion of, or provide edits to an open RFC
+- Discuss the Rust project in an official chat (internals.rust-lang.org, rust-lang Zulip, etc.)
+- Open an issue on any project in the rust-lang GitHub organization
+- Contribute code changes (including tests) to any project in the rust-lang GitHub organization
+- Contribute non-code changes (documentation, comments, etc.) to any project in the rust-lang GitHub organization
+
+Frequency:
+- More frequently than weekly
+- Weekly
+- Monthly or less frequently
+- Never
+
+> **justification**
+>
+> We want to understand the nature of contribution to the Rust project both
+> to better understand the shape of community involvement and for cohort analysis.
+
+### How do you feel in the Rust community?
+
+Type: matrix
+
+Activities:
+
+- *Official* Rust community forums or chats (users.rust-lang.org, internals.rust-lang.org, the official Rust Discord, or the Rust Zulip)
+- *Unofficial* Rust community forums or chats (e.g., reddit.com/r/rust, Hacker News, the Rust *Community* Discord, etc.)
```

Comment:
Perhaps we should add communities for Rust crates? Or add another category for it?

Reply:
I'm not sure I understand the suggestion :slightly_smiling_face:  I see crates as an _ecosystem_ rather than a community (iiuc what you mean)

Reply:
So, I help run [Bevy](https://bevyengine.org/). We have a thriving independent community of thousands of Rust developers. For most of them, this is their primary interaction with the Rust community.

Similar communities exist for e.g. Dimforge, Rust Audio, AeroRust and so on.

Reply:
totally fair point, thanks for clarifying. I suggest the following wording:

> * Community focused on a specific area of software development

wdyt @alice-i-cecile ? feel free to suggest :slightly_smiling_face:

Reply:
Mention "Rust" in there somehow and it LGTM.

Reply:
I added `- Community focused on a specific area of Rust software development (e.g. game development, audio, etc.)`. We can change it later of course.